guides/release/components/actions-and-events.md

```diff
@@ -0,0 +1,609 @@
+import Service from '@ember/service';
+import { action } from '@ember/object';
+
+export default class Messaging extends Service {
+  @action
+  sendMessage(messageType, text) {
+    // handle message send and return a promise
+  }
+}
+```
+
+## Destructuring Objects Passed as Action Arguments
+
+A component will often not know what information a parent needs to process an
+action, and will just pass all the information it has. For example, our
+`UserProfile` component is going to notify its parent,
+`system-preferences-editor`, that a user's account was deleted, and passes along
+with it the full user profile object.
+
+```javascript {data-filename=app/components/user-profile.js}
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+
+export default class UserProfile extends Component {
+  @service
+  login;
+
+  @action
+  userDidDeleteAccount() {
+    this.login.deleteUser();
+    this.didDelete(this.login.currentUserObj);
+  }
+}
+```
+
+All our `system-preferences-editor` component really needs to process a user
+deletion is an account ID. For this case, the action helper provides the `value`
+attribute to allow a parent component to dig into the passed object to pull out
+only what it needs.
+
+```handlebars {data-filename=app/templates/components/system-preferences-editor.hbs}
+<UserProfile @didDelete={{action this.userDeleted value="account.id"}} />
+```
+
+Now when the `system-preferences-editor` handles the delete action, it receives
+only the user's account `id` string.
+
+```javascript {data-filename=app/components/system-preferences-editor.js}
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class SystemPreferencesEditor extends Component {
+  @action
+  userDeleted(idStr) {
+    // respond to deletion
+  }
+}
+```
+
+## Calling Actions Up Multiple Component Layers
+
+When your components go multiple template layers deep, it is common to need to
+handle an action several layers up the tree. Using the action helper, parent
+components can pass actions to child components through templates alone without
+adding JavaScript code to those child components.
+
+For example, say we want to move account deletion from the `UserProfile`
+component to its parent `system-preferences-editor`.
+
+First we would move the `deleteUser` action from `user-profile.js` to the
+actions object on `system-preferences-editor`.
+
+```javascript {data-filename=app/components/system-preferences-editor.js}
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+
+export default class SystemPreferencesEditor extends Component {
+  @service
+  login;
+
+  @action
+  deleteUser(idStr) {
+    return this.login.deleteUserAccount(idStr);
+  }
+}
+```
+
+Then our `system-preferences-editor` template passes its local `deleteUser`
```

Comment:
I'd avoid the user of "local" here since it clashes with the description of "local state" in the state management section